core/api/user/router.py

```diff
@@ -1,45 +1,45 @@
 from fastapi import APIRouter, Depends, HTTPException
-from sqlalchemy.orm import Session
+from sqlalchemy.ext.asyncio import AsyncSession
 
-from database import crud
+from core.domain.user import crud
 from database.engine import get_db
-from database.exceptions import EmailAlreadyExistsError
-from database.schemas import User, UserCreate, UserUpdate
+from core.domain.user.exceptions import EmailAlreadyExistsError
+from core.domain.user.schemas import User, UserCreate, UserUpdate
 
 router = APIRouter()
 
 
 @router.get("/users/", response_model=list[User])
-def read_users(skip: int = 0, limit: int = 10, db: Session = Depends(get_db)):
-    return crud.get_users(db=db, skip=skip, limit=limit)
+async def read_users(skip: int = 0, limit: int = 10, db: AsyncSession = Depends(get_db)):
+    return await crud.get_users(db=db, skip=skip, limit=limit)
```

Comment:
возвращаемый тип данных в роуте отсутствует. Понимаю, смысла в нем супер особо нет, но это просто бестпрактис и муру такое не пропускает.
модель алхимии возвращать не стоит и таскать через слои (в нашем случае перемещение по слоям - это функция круд -> роут). Бест практис - круд отработал, а в месте где вызывает это круд (в нашем случае роут) заворачиваем в pydantic model, то есть надо создать pydantic model, например `UserSchema` с полями, которые будем возвращать завернуть и вернуть
```suggestion
async def read_users(skip: int = 0, limit: int = 10, db: AsyncSession = Depends(get_db)) -> list[UserSchema]:
    result = await crud.get_users(db=db, skip=skip, limit=limit)
    return UserSchema(
        id=result.id,
        name=result.name,
        etc...
    )
```